fairlearn/preprocessing/_information_filter.py

```diff
@@ -0,0 +1,123 @@
+
+    :param columns: list of columns to filter out this can be a sequence of either int
+                    (in the case of numpy) or string (in the case of pandas).
+    :param alpha: parameter to control how much to filter, for alpha=1 we filter out
+                  all information while for alpha=0 we don't apply any.
+    :param center: setting to tell if this preprocessing step should center the data for
+                   numerical stability
+    """
+
+    def __init__(self, columns, alpha=1, center=True):
+        self.columns = columns
+        self.alpha = alpha
+        self.center = center
+
+    def _check_coltype(self, X):
+        check_array(X, estimator=self)
+        for col in self.columns:
+            if isinstance(col, str):
+                if isinstance(X, np.ndarray):
+                    raise ValueError(
+                        f"column {col} is a string but datatype received is numpy."
+                    )
+                if isinstance(X, pd.DataFrame):
+                    if col not in X.columns:
+                        raise ValueError(f"column {col} is not in {X.columns}")
+            if isinstance(col, int):
+                if col not in range(np.atleast_2d(np.array(X)).shape[1]):
+                    raise ValueError(
+                        f"column {col} is out of bounds for input shape {X.shape}"
+                    )
+
+    def _col_idx(self, X, name):
+        if isinstance(name, str):
+            if isinstance(X, np.ndarray):
+                raise ValueError(
+                    "You cannot have a column of type string on a numpy input matrix."
+                )
+            return {name: i for i, name in enumerate(X.columns)}[name]
+        return name
+
+    def _make_v_vectors(self, X, col_ids):
+        vs = np.zeros((X.shape[0], len(col_ids)))
+        for i, c in enumerate(col_ids):
+            vs[:, i] = X[:, col_ids[i]]
+            for j in range(0, i):
+                vs[:, i] = vs[:, i] - _vector_projection(vs[:, i], vs[:, j])
+        return vs
+
+    def fit(self, X, y=None):
+        """Learn the projection required to make the dataset orthogonal to sensitive columns."""
+        self._check_coltype(X)
+        self.col_ids_ = [v if isinstance(v, int) else self._col_idx(X, v) for v in self.columns]
+        X = check_array(X, estimator=self)
+        if self.center:
+            X = StandardScaler(with_std=False).fit_transform(X)
+        X_fair = X.copy()
+        v_vectors = self._make_v_vectors(X, self.col_ids_)
+        # Gram-Schmidt process but only on sensitive attributes
+        for i, col in enumerate(X_fair.T):
+            for v in v_vectors.T:
+                X_fair[:, i] = X_fair[:, i] - _vector_projection(X_fair[:, i], v)
+        # We want to learn matrix P: X P = X_filtered
+        # This means we first need to create X_filtered in order to learn P
+        self.projection_, resid, rank, s = np.linalg.lstsq(X, X_fair, rcond=None)
+        return self
```

Comment:
I don't believe we've done this on our other `fit()` methods... I'm guessing it allows users to do things like `est.fit(X).predict(X_test)` ?

Reply:
Exactly, this is also what for example sklearns `TransformerMixin` relies on:

```
class TransformerMixin:
    ....
    def fit_transform(self, X, y):
        ....
        return self.fit(X, **fit_params).transform(X)```
```